config/autoload/global.php

```diff
@@ -24,7 +24,13 @@
         ),
         'from' => 'secr@gewis.nl',
         'to' => [
-            'report_error' => 'databaseisstuk@gewis.nl'
+            'report_error' => 'databaseisstuk@gewis.nl',
+            'subscription' => 'secr@gewis.nl'
         ]
     ),
+    'storage' => array(
+        'storage_dir' => 'public/data',
+        'public_dir' => 'data',
+        'dir_mode' => 0777,
```

Comment:
No need om dit 777 op de GEWIS Server te zetten. `public` is voor bestanden die secr-public zijn, niet GEWIS-public.

We zitten hier eigenlijk met iets kuts, want we gaan handtekeningen opslaan. Normaalgezien zou je al deze bestanden via de webserver (apache) willen serveren aan de secr. Maar dat kunnen we niet maken, gezien met een beetje trucerij (lees: `.htaccess` aanpassen) actieve leden hier ook bij zouden kunnen. Dus hier gaan we wat anders op moeten verzinnen.

Mijn idee zou zijn ze in de database op te slaan als base64, en met behulp van `<img src="data:image/jpeg;base64 [...]` ze te laten zien.

Reply:
@tomudding Maybe you can comment on this since you created this feature?

Reply:
@kokx Initieel had ik inderdaad een base64 data URI in de database staan, maar zoals te zien is heb ik dit veranderd. Ik test misschien iets te uitgebreid, maar tijdens het testen merkte ik dat het mogelijk is om de data URI op te blazen. Hierdoor ontstonden er problemen met de maximale lengte van een `TEXT` attribuut in de database.

Tevens heb ik na zitten denken over de kans dat mensen die de foto's niet horen te zien deze wel kunnen zijn. Via de website is toegang vrij lastig (gezien de gigantische range van de hashes), maar even over het hoofd gezien kan je natuurlijk ook actieve leden hebben die de boel aan het verpesten zijn.

Het beste zou dan inderdaad het opslaan van de data URI zijn, maar dan zou ik deze wel inkorten als hij te lang is (TEXT is dacht ik 64Kb). In de hoop dat men niet te veel loopt te veel met het form aan het spelen is zou de data URI met de huidige code tussen de 10Kb en 20Kb moeten zijn.

@Koen1999 Ik wil het wel voor je aanpassen, maar je kan het zelf ook veranderen (houd wel rekening met mogelijke stored XSS).

Reply:
@tomudding @kokx Oeps, ik zie dit ook een beetje laat. Als je het zou willen aanpassen zou dat fijn zijn. Zo niet lijkt het me ook niet heel erg problematisch om het zo te laten en 777 aan te passen naar bijvoorbeeld 660. De files worden iig al verwijderd zodra het lid wordt omgezet tot echt lid. En het is redelijk lastig om te gokken achter welke URL een handtekening zit (tenzij je kijkt welke URLs bezocht worden)